staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -181,7 +184,7 @@ func (t *watcherBookmarkTimeBuckets) addWatcher(w *cacheWatcher) bool {
 	if !ok {
 		return false
 	}
-	bucketID := nextTime.Unix()
+	bucketID := nextTime.Sub(t.createTime) / time.Second
```

Comment:
Can we add a unit test to validate that we're no longer leaking watchers?

Reply:
Yeah, I think that's a good idea. I'll add a unit test.

Reply:
I think modifying the system time in unit test is not a good idea, there is currently no good idea to test this scenario

Reply:
I think modifying the system time in unit test is not a good idea, there is currently no good idea to test this scenario